icu4c/source/i18n/units_complexconverter.cpp

```diff
@@ -177,14 +201,25 @@ MaybeStackVector<Measure> ComplexUnitsConverter::convert(double quantity,
         }
     }
 
-    // Package values into Measure instances in result:
-    for (int i = 0, n = unitConverters_.length(); i < n; ++i) {
+    struct MeasureWithIndex {
+        const int32_t index;
+        const Measure measure;
+        MeasureWithIndex(int32_t index,const Measure &measure) : index(index), measure(measure) {}
+    };
+
+    // Package values into Measure instances in unordered_result:
+    // FIXME/optional: I'm wondering if unordered_result is necessary at all, when the
```

Comment:
Optional: I added this fixme while reviewing/editing the code in vscode - it might be unnecessary to have the unordered_result variable.

Reply:
why, we need to sort them. Therefore, we need the index with Measure

Reply:
Ah... I'm sorry, I don't know how I missed it on Friday (I can speculate), but I somehow managed to miss that you're restoring the original order by sorting again. Having not noticed the second sort, I was looking for the logic that simply puts the values back at their required indexes. (You do this in Java, I think? Not sorting twice?)

So how would you like to do this? Leave Java and C++ implementations different, because this does indeed behave correctly?

Reply:
Done: I removed the second sort.